### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by piyadaswsa**

### CODE REVIEW

File: app/src/main/res/layout/item_fav_mobile.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            app:layout_constraintStart_toEndOf="@+id/iv_phone"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <TextView
+            android:id="@+id/tv_phone_price"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/margin_4"
+            android:layout_weight="8"
+            app:layout_constraintStart_toStartOf="@+id/tv_phone_name"
```
Comment (Bugchain) [inline app/src/main/res/layout/item_fav_mobile.xml]: Use dimens resource.

File: app/src/main/res/layout/item_mobile.xml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="@+id/tv_phone_name"
+            app:layout_constraintTop_toBottomOf="@+id/tv_phone_name" />
+
+        <TextView
+            android:id="@+id/tv_phone_price"
+            android:layout_width="wrap_content"
+            android:layout_height="@dimen/tv_phone_price_height_size"
+            android:layout_marginTop="@dimen/margin_8"
+            android:layout_weight="8"
```
Comment (Bugchain) [inline app/src/main/res/layout/item_mobile.xml]: Use dimen resource.

File: app/src/main/java/com/example/mobilephone/adapter/ImageAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.example.mobilephone.adapter
+
+import android.content.Context
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.mobilephone.model.MobileImage
+import com.example.mobilephone.viewholder.MobileImageViewHolder
+
+class ImageAdapter(private var mobileImages: List<MobileImage>, private var context: Context) :
+    RecyclerView.Adapter<MobileImageViewHolder>() {
```
Comment (Bugchain) [inline app/src/main/java/com/example/mobilephone/adapter/ImageAdapter.kt]: If variable not change please use val.

**Comment by piyadaswsa** [comment]
Rewrite code style and Fix code from P'Chain's review

File: app/src/main/java/com/example/mobilephone/viewholder/FavoriteMobileViewHolder.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import androidx.recyclerview.widget.RecyclerView
+import com.example.mobilephone.R
+import com.example.mobilephone.extension.setImageUrl
+import com.example.mobilephone.model.Mobile
+
+class FavoriteMobileViewHolder(parent: ViewGroup) : RecyclerView.ViewHolder(
+    LayoutInflater.from(parent.context).inflate(R.layout.item_fav_mobile, parent, false)
+) {
+
+    private val ivMobileImg: ImageView = itemView.findViewById(R.id.iv_phone)
```
Comment (last3oy) [inline app/src/main/java/com/example/mobilephone/viewholder/FavoriteMobileViewHolder.kt]: Once you are using kotlin synthetic please apply this to kotlin synthetic as well.

File: app/src/main/java/com/example/mobilephone/viewholder/MobileImageViewHolder.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import com.example.mobilephone.R
+import com.example.mobilephone.extension.setImageUrl
+import com.example.mobilephone.model.MobileImage
+
+class MobileImageViewHolder(parent: ViewGroup, private var context: Context) :
+    RecyclerView.ViewHolder(
+        LayoutInflater.from(parent.context).inflate(R.layout.item_image_mobile, parent, false)
+    ) {
+
+    private val ivMobileImg: ImageView = itemView.findViewById(R.id.iv_image_mobile)
```
Comment (last3oy) [inline app/src/main/java/com/example/mobilephone/viewholder/MobileImageViewHolder.kt]: use kotlin synthetic

File: app/src/main/java/com/example/mobilephone/viewholder/MobileItemViewHolder.kt [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+import com.example.mobilephone.extension.OnMobileClickListener
+import com.example.mobilephone.extension.setImageUrl
+import com.example.mobilephone.model.Mobile
+
+class MobileItemViewHolder(var parent: ViewGroup, var context: Context) : RecyclerView.ViewHolder(
+    LayoutInflater.from(parent.context).inflate(R.layout.item_mobile, parent, false)
+) {
+
+    private val ivMobileImg: ImageView = itemView.findViewById(R.id.iv_phone)
+    private val ivMobileFav: ImageView = itemView.findViewById(R.id.iv_fav)
```
Comment (last3oy) [inline app/src/main/java/com/example/mobilephone/viewholder/MobileItemViewHolder.kt]: kotlin synthetic

File: app/src/main/res/layout/activity_mobile_detail.xml [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+            android:id="@+id/progress_bar_in_detail"
+            layout="@layout/progress"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            tools:visibility="visible" />
+
+    </LinearLayout>
+</ScrollView>
+
+
```
Comment (last3oy) [inline app/src/main/res/layout/activity_mobile_detail.xml]: Remove line spaces

**Pull request merged**